### PR TITLE
Use new versioner to calculate existing v1 lock file

### PR DESCRIPTION
**Pull request by danielrbradley**
There's probably more places we could simplify the v1 config to use tracking without affecting the v1.json lock file but need to experiment more here.

### CODE REVIEW

File: provider/cmd/pulumi-versioner-azure-native/main.go [hunk 1]
```diff
@@ -183,23 +183,23 @@ func pending(outputDir, versionFile string) error {
...
 
-	v1 := openapi.CalculateProviderDefaults(activePathVersions, providers)
+	v2, err := versioning.DefaultConfigToCuratedVersion(specVersions, v2Config)
 	if err != nil {
 		return err
 	}
 
 	return emitFiles(outputDir, map[Filename]Data{
-		"v1.json": v1,
+		"v1.json": v2,
```
Comment (stack72) [inline provider/cmd/pulumi-versioner-azure-native/main.go:202]: Why are we writing v2 config to a v1 json file?

Comment (danielrbradley) [reply provider/cmd/pulumi-versioner-azure-native/main.go:202]: Copy pasta error - didn't rename the variable .. it is v1 🤦

File: Makefile [hunk 2]
```diff
@@ -86,7 +86,8 @@ versions/spec-resources.json: bin/pulumi-versioner-azure-native .git/modules/azu
 versions/active.json: bin/pulumi-versioner-azure-native azure-provider-versions/provider_list.json
 	bin/pulumi-versioner-azure-native active
 
-versions/v1.json: bin/pulumi-versioner-azure-native .git/modules/azure-rest-api-specs/HEAD azure-provider-versions/provider_list.json
+V1CONFIG := $(wildcard versions/v1-config.yaml)
```
Comment (viveklak) [inline Makefile:89]: Do we need the wildcard expansion here or are we planning to add one in the future?

Comment (danielrbradley) [reply Makefile]: Nope, removed!

**Review by viveklak — APPROVED** [review]

**Pull request merged**